### ISSUE TITLE
🐛 Crashes when displaying ad banners in Listview on iOS

### ISSUE THREAD

**Issue by KevinCCucumber** [googleads/googleads-mobile-flutter#159]
## Bug report

When using adbanners every x entrys in a Listview, flutter crashes on iOS

### Steps to reproduce

1. Implement Banner Ad in a Container
2. Call the Container within a Listview.separated
3. App crashes as soon as the first ad banner enters the viewing range

### Expected behavior

The List scrolls down with an adbanner showing every x entries

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.18363.1440], locale de-DE)
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Android Studio (version 3.6)
[√] VS Code (version 1.55.0)
[√] Connected device (1 available)
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.2
Flutter SDK 2.0.4
flutter_ac 2.6.10+116

dependencies:
- cupertino_icons 0.1.3
- device_info 0.4.2+10 [flutter device_info_platform_interface]
- expandable 4.1.4 [flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_styled_toast 1.4.0+1 [flutter flutter_localizations]
- google_mobile_ads 0.11.0+4 [meta flutter]
- http 0.12.2 [http_parser path pedantic]
- package_info 0.4.3+2 [flutter]
- path_provider 1.6.24 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- path_provider_ex 1.0.1 [flutter]
- provider 4.3.2+2 [flutter nested collection]
- shared_preferences 0.5.12+4 [meta flutter shared_preferences_platform_interface shared_preferences_linux shared_preferences_macos shared_preferences_web shared_preferences_windows]
- sqflite 1.3.1+2 [flutter sqflite_common path]
- uni_links 0.4.0 [flutter]
- url_launcher 5.7.10 [flutter url_launcher_platform_interface url_launcher_web url_launcher_linux url_launcher_macos url_launcher_windows]
- webview_flutter 1.0.6 [flutter]

dev dependencies:
- flutter_launcher_icons 0.7.5 [image args yaml]

transitive dependencies:
- archive 2.0.13 [crypto args path]
- args 1.6.0
- characters 1.1.0
- charcode 1.1.3
- clock 1.1.0
- collection 1.15.0
- convert 2.1.1 [charcode typed_data]
- crypto 2.1.5 [collection convert typed_data]
- device_info_platform_interface 1.0.1 [flutter meta plugin_platform_interface]
- ffi 0.1.3
- file 6.1.0 [meta path]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- image 2.1.18 [archive xml meta]
- intl 0.17.0 [clock path]
- js 0.6.3
- meta 1.3.0
- nested 0.0.4 [flutter]
- path 1.8.0
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+6 [flutter]
- path_provider_platform_interface 1.0.4 [flutter meta platform plugin_platform_interface]
- path_provider_windows 0.0.4+3 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.9.2 [meta]
- petitparser 3.1.0 [meta]
- platform 3.0.0
- plugin_platform_interface 1.0.3 [meta]
- process 4.2.1 [file path platform]
- shared_preferences_linux 0.0.2+4 [file flutter meta path path_provider_linux shared_preferences_platform_interface]
- shared_preferences_macos 0.0.1+11 [shared_preferences_platform_interface flutter]
- shared_preferences_platform_interface 1.0.4 [meta flutter]
- shared_preferences_web 0.1.2+7 [shared_preferences_platform_interface flutter flutter_web_plugins meta]
- shared_preferences_windows 0.0.1+3 [shared_preferences_platform_interface flutter ffi file meta path path_provider_platform_interface path_provider_windows]
- sky_engine 0.0.99
- source_span 1.7.0 [charcode collection meta path term_glyph]
- sqflite_common 1.0.2+1 [synchronized path meta]
- string_scanner 1.0.5 [charcode meta source_span]
- synchronized 2.2.0+2
- term_glyph 1.1.0
- typed_data 1.3.0 [collection]
- url_launcher_linux 0.0.1+4 [flutter]
- url_launcher_macos 0.0.1+9 [flutter]
- url_launcher_platform_interface 1.0.9 [flutter meta plugin_platform_interface]
- url_launcher_web 0.1.5+1 [url_launcher_platform_interface flutter flutter_web_plugins meta]
- url_launcher_windows 0.0.1+3 [flutter]
- vector_math 2.1.0
- win32 1.7.3 [ffi]
- xdg_directories 0.1.2 [meta path process]
- xml 4.5.1 [collection convert meta petitparser]
- yaml 2.2.1 [charcode collection string_scanner source_span]
```

</details>

---

**Comment by mustafa-707** [comment]
same here , looking for any solution , it happened with me after upgrading to flutter 2

**Comment by jjliu15** [comment]
Can you share the crash logs?

**Comment by mustafa-707** [comment]
solved for me after flutter 2.2.0 update
